### ISSUE TITLE
Unwanted verbose output about nix fish init scripts when starting fish

### ISSUE THREAD

**Issue by tabbyrobin** [NixOS/nix#7522]
**Describe the bug**

After installing nix, there is verbose output whenever I start fish shell.  I'm not sure whether these are errors or just verbose output. Generally nix seems to work fine. This is what shows at every launch:

```shell
/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.fish (line 34): 
fish_add_path --prepend --global "/nix/var/nix/profiles/default/bin"
^
from sourcing file /nix/var/nix/profiles/default/etc/profile.d/nix-daemon.fish
	called on line 185 of file /usr/share/fish/config.fish
in function '.' with arguments '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.fish'
	called on line 4 of file /etc/fish/conf.d/nix.fish
from sourcing file /etc/fish/conf.d/nix.fish
	called on line 294 of file /usr/share/fish/config.fish
from sourcing file /usr/share/fish/config.fish
	called during startup
/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.fish (line 35): 
fish_add_path --prepend --global "$HOME/.nix-profile/bin"
^
from sourcing file /nix/var/nix/profiles/default/etc/profile.d/nix-daemon.fish
	called on line 185 of file /usr/share/fish/config.fish
in function '.' with arguments '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.fish'
	called on line 4 of file /etc/fish/conf.d/nix.fish
from sourcing file /etc/fish/conf.d/nix.fish
	called on line 294 of file /usr/share/fish/config.fish
from sourcing file /usr/share/fish/config.fish
	called during startup
Welcome to fish, the friendly interactive shell
Type `help` for instructions on how to use fish
```

**Steps To Reproduce**

1. Have fish installed on debian 10.
2. Install nix multiuser via `sh <(curl -L https://nixos.org/nix/install) --daemon` as per https://nixos.org/download.html (Yes to use sudo)
3. After install succeeds, restart shells.
4. Start fish, see verbose output before fish prompt appears.

**Expected behavior**

Same as observed behavior, except without the verbose output when starting fish.

**`nix-env --version` output**

nix-env (Nix) 2.12.0

**Additional context**

I am using a qubes standalone vm based on the debian-10 template.

**Comment by abathur** [comment]
I think this is the same issue reported in #7546

**Comment by edolstra** [comment]
Should be fixed by #7547.
